web/job_runs_controller.go

```diff
@@ -55,6 +63,20 @@ func (jrc *JobRunsController) Create(c *gin.Context) {
 	}
 }
 
+func getRunData(c *gin.Context) (models.JSON, error) {
+	data := models.JSON{}
+	b, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		return data, err
+	}
+	if len(b) == 0 {
+		c.Request.Body = ioutil.NopCloser(bytes.NewBufferString(`{}`))
```

Comment:
I tried to make this work in models.JSON`s `UnmarshalJSON` but gin's ShouldBindJSON still had problems. I think that'd still be worth doing in the future(for easier `runlog`) but for now I figured we can just handle it here.

Reply:
So I don't think you even need `c.ShouldBindJSON`. Can we just deserialize it ourselves into `JSON`? What does `c.ShouldBindJSON` give over the following:

```golang
func getRunData(c *gin.Context) (models.JSON, error) {
var data models.JSON
b, err := ioutil.ReadAll(c.Request.Body) 
if err != nil {
	return data, err
}
err = json.Marshal(b, &data)
return data, err
}
```

Reply:
Ok, so turns out we can't push it into JSON.UnmarshalJSON, because json.Unmarshal still does a few [validation checks which an empty string doesn't pass](https://github.com/golang/go/blob/master/src/encoding/json/decode.go#L101).